src/main/java/net/dries007/tfc/compat/wailatop/WailaBlocks.java

```diff
@@ -0,0 +1,502 @@
+        registrar.registerHeadProvider(dataProvider, BlockFruitTreeTrunk.class);
+        registrar.registerHeadProvider(dataProvider, BlockFruitTreeBranch.class);
+
+        //Body
+        registrar.registerBodyProvider(dataProvider, BlockOreTFC.class);
+        registrar.registerBodyProvider(dataProvider, TEPitKiln.class);
+        registrar.registerBodyProvider(dataProvider, TEFirePit.class);
+        registrar.registerBodyProvider(dataProvider, TEBloomery.class);
+        registrar.registerBodyProvider(dataProvider, TEBlastFurnace.class);
+        registrar.registerBodyProvider(dataProvider, TECharcoalForge.class);
+        registrar.registerBodyProvider(dataProvider, TEBarrel.class);
+        registrar.registerBodyProvider(dataProvider, TELogPile.class);
+        registrar.registerBodyProvider(dataProvider, TEPlacedItemFlat.class);
+
+        registrar.registerBodyProvider(dataProvider, TECropBase.class);
+        registrar.registerBodyProvider(dataProvider, BlockFarmlandTFC.class);
+        registrar.registerBodyProvider(dataProvider, BlockFruitTreeLeaves.class);
+        registrar.registerBodyProvider(dataProvider, BlockBerryBush.class);
+        registrar.registerBodyProvider(dataProvider, BlockCropDead.class);
+
+        //Tail
+        registrar.registerTailProvider(dataProvider, BlockRockVariant.class);
+
+
+    }
+
+    @Override
+    public ItemStack getWailaStack(IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        ItemStack itemstack = ItemStack.EMPTY;
+
+        if (b instanceof BlockOreTFC)
+        {
+            itemstack = OreTFCStack(accessor, config);
+        }
+        else if (b instanceof BlockCropSimple)
+        {
+            itemstack = CropSimpleStack(accessor, config);
+        }
+        return itemstack;
+    }
+
+    @Override
+    public List<String> getWailaHead(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
+    {
+        Block b = accessor.getBlock();
+        TileEntity te = accessor.getTileEntity();
+
+        if (b instanceof BlockOreTFC) currenttip = OreTFCHead(itemStack, currenttip, accessor, config);
+        if (b instanceof BlockBarrel) currenttip = BarrelHead(itemStack, currenttip, accessor, config);
+        if (te instanceof TECropBase) currenttip = CropSimpleHead(itemStack, currenttip, accessor, config);
+        if (b instanceof BlockCropDead) currenttip = CropDeadHead(itemStack, currenttip, accessor, config);
+        if (b instanceof BlockFruitTreeTrunk | b instanceof BlockFruitTreeLeaves | b instanceof BlockFruitTreeBranch)
+        {
+            currenttip = FruitTreeHead(itemStack, currenttip, accessor, config);
+        }
+        return currenttip;
+    }
+
+    @Override
+    public List<String> getWailaBody(ItemStack itemStack, List<String> currenttip, IWailaDataAccessor accessor, IWailaConfigHandler config)
```

Comment:
```suggestion
    public List<String> getWailaBody(ItemStack stack, List<String> currentTooltip, IWailaDataAccessor accessor, IWailaConfigHandler config)
```
As per both java conventions on field / paramater naming, and MCP coding style conventions